applications/mne_anonymize/fiffanonymizer.cpp

```diff
@@ -0,0 +1,724 @@
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace MNEANONYMIZE;
+using namespace FIFFLIB;
+
+
+//*************************************************************************************************************
+//========================================================= ====================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+FiffAnonymizer::FiffAnonymizer()
+: version(1.0)
+, maxValidFiffVerion(1.3)
+, versionStr(QString::number(version))
+, name("MNE Anonymize")
+, description("Application that removes or modifies Personal Health Information or Personal Identifiable information.")
+, m_bVerboseMode(false)
+, m_bBruteMode(false)
+, m_bQuietMode(false)
+, m_bDeleteInputFileAfter(false)
+, m_bDeleteInputFileConfirmation(true)
+, m_sDfltString("mne_anonymize")
+, m_dateDfltDate(QDateTime(QDate(2000,1,1), QTime(1, 1, 0)))
+, m_iMeasurementDayOffset(0)
+, m_bUseMeasurementDayOffset(false)
+, m_iSubjectBirthdayOffset(0)
+, m_bUseSubjectBirthdayOffset(false)
+, m_dateMeasurmentDate(m_dateDfltDate)
+, m_date_IdDate(m_dateDfltDate)
+, m_iDfltSubjectId(0)
+, m_sDfltSubjectFirstName(m_sDfltString)
+, m_sDfltSubjectMidName("x")
+, m_sDfltSubjectLastName(m_sDfltString)
+, m_dateSubjectBirthDay(m_dateDfltDate)
+, m_iDfltSubjectWeight(0)
+, m_iDfltSubjectHeight(0)
+, m_sDfltSubjectComment(m_sDfltString)
+, m_sDfltSubjectHisId(m_sDfltString)
+, m_iDfltProjectId(0)
+, m_sDfltProjectName(m_sDfltString)
+, m_sDfltProjectAim(m_sDfltString)
+, m_sDfltProjectPersons(m_sDfltString)
+, m_sDfltProjectComment(m_sDfltString)
+, m_sFileNameIn("")
+, m_sFfileNameOut("")
+, m_printInSameLineHelper(qDebug())
+, m_bPrintInSameLine(true)
+{
+    m_BDfltMAC.resize(8);
+    m_BDfltMAC[0] = 0x00;
+    m_BDfltMAC[1] = 0x00;
+    m_BDfltMAC[2] = 0x00;
+    m_BDfltMAC[3] = 0x00;
+    m_BDfltMAC[4] = 0x00;
+    m_BDfltMAC[5] = 0x00;
+    m_BDfltMAC[6] = 0x00;
+    m_BDfltMAC[7] = 0x00;
+
+    m_pBlockTypeList = QSharedPointer<QStack<int32_t> >(new QStack<int32_t>);
+    m_pOutDir = QSharedPointer<QVector<FiffDirEntry> >(new QVector<FiffDirEntry>);
+}
+
+
+//*************************************************************************************************************
+
+FiffAnonymizer::FiffAnonymizer(const FiffAnonymizer& obj)
```

Comment:
This is just a first version of the copy constructor. We need get methods next to all set methods to retreive each to be copied parameter of obj here.

Reply:
Thanks for your last pushes. Yes. I'll try to do this today.